moveit_ros/planning_interface/test/planning_scene_interface_clearScene.cpp

```diff
@@ -0,0 +1,207 @@
+
+  int object_counter = 0;
+
+  moveit_msgs::CollisionObject randomCollisionObject()
+  {
+    std::random_device rd;   // Seed the engine with a true random value if available
+    std::mt19937 gen(rd());  // Mersenne Twister 19937 generator
+    std::uniform_real_distribution<float> dist(-4.0f, 4.0f);
+
+    moveit_msgs::CollisionObject collision_object;
+    collision_object.id = std::to_string(object_counter);
+    collision_object.header.frame_id = "panda_link0";
+    collision_object.primitives.resize(1);
+    collision_object.primitives[0].type = collision_object.primitives[0].BOX;
+    collision_object.primitives[0].dimensions.resize(3);
+    collision_object.primitives[0].dimensions[0] = std::abs(dist(gen));
+    collision_object.primitives[0].dimensions[1] = std::abs(dist(gen));
+    collision_object.primitives[0].dimensions[2] = std::abs(dist(gen));
+    collision_object.primitive_poses.resize(1);
+    collision_object.primitive_poses[0].position.x = dist(gen);
+    collision_object.primitive_poses[0].position.y = dist(gen);
+    collision_object.primitive_poses[0].position.z = dist(gen);
+    collision_object.primitive_poses[0].orientation.w = 1.0;
+    collision_object.operation = collision_object.ADD;
+
+    // Global random counter update
+    object_counter++;
+
+    return collision_object;
+  }
+  moveit_msgs::AttachedCollisionObject randomAttachedCollisionObject()
+  {
+    std::random_device rd;   // Seed the engine with a true random value if available
+    std::mt19937 gen(rd());  // Mersenne Twister 19937 generator
+    std::uniform_real_distribution<float> dist(-4.0f, 4.0f);
+
+    moveit_msgs::AttachedCollisionObject attached_collision_object;
+    attached_collision_object.link_name = "panda_link0";
+    attached_collision_object.object.id = std::to_string(object_counter);
+    attached_collision_object.object.header.frame_id = "panda_link0";
+    attached_collision_object.object.primitives.resize(1);
+    attached_collision_object.object.primitives[0].type = attached_collision_object.object.primitives[0].BOX;
+    attached_collision_object.object.primitives[0].dimensions.resize(3);
+    attached_collision_object.object.primitives[0].dimensions[0] = std::abs(dist(gen));
+    attached_collision_object.object.primitives[0].dimensions[1] = std::abs(dist(gen));
+    attached_collision_object.object.primitives[0].dimensions[2] = std::abs(dist(gen));
+    attached_collision_object.object.primitive_poses.resize(1);
+    attached_collision_object.object.primitive_poses[0].position.x = dist(gen);
+    ;
+    attached_collision_object.object.primitive_poses[0].position.y = dist(gen);
+    ;
+    attached_collision_object.object.primitive_poses[0].position.z = dist(gen);
+    ;
+    attached_collision_object.object.primitive_poses[0].orientation.w = 1.0;
+    attached_collision_object.object.operation = attached_collision_object.object.ADD;
+
+    // Global random counter update
+    object_counter++;
```

Comment:
This can be further simplified:
```suggestion
 attached_collision_object.object = randomCollisionObject();
```

Reply:
I don't know why I couldn't see that, this makes so much more sense.